nft_ingester/src/bin/synchronizer/main.rs

```diff
@@ -120,34 +120,88 @@ pub async fn main() -> Result<(), IngesterError> {
         metrics.clone(),
         config.parallel_tasks,
         config.run_temp_sync_during_dump,
-    );
+    ));
 
     if let Err(e) = rocks_storage.db.try_catch_up_with_primary() {
         tracing::error!("Sync rocksdb error: {}", e);
     }
-    synchronizer
-        .maybe_run_full_sync(&shutdown_rx, config.dump_sync_threshold)
-        .await;
 
-    while shutdown_rx.is_empty() {
-        if let Err(e) = rocks_storage.db.try_catch_up_with_primary() {
-            tracing::error!("Sync rocksdb error: {}", e);
-        }
-        let res = synchronizer
-            .synchronize_asset_indexes(&shutdown_rx, config.dump_sync_threshold)
-            .await;
-        match res {
-            Ok(_) => {
-                tracing::info!("Synchronization finished successfully");
+    for asset_type in ASSET_TYPES {
+        let synchronizer = synchronizer.clone();
+        let shutdown_rx = shutdown_rx.resubscribe();
+
+        mutexed_tasks.lock().await.spawn(async move {
+            if let Ok(SyncStatus::FullSyncRequired(_)) = synchronizer
+                .get_sync_state(config.dump_sync_threshold, asset_type)
+                .await
+            {
+                let res = synchronizer.full_syncronize(&shutdown_rx, asset_type).await;
```

Comment:
This is exactly the `maybe_run_full_sync` function, `full_syncronize` will not be called if `FullSync` isn't required. I erased the function because it does nothing except for logging + cannot be called from different threads because of passing &self which doesn't have Send + Sync by default.